lib/src/utf8.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library utf8;
+
+import 'dart:convert';
+import 'dart:ffi';
+import 'dart:typed_data';
+
+/// [Utf8] implements conversion between Dart strings and null-termianted
+/// Utf8-encoded "char*" strings in C.
+class Utf8 extends Struct<Utf8> {
+  static String fromUtf8(Pointer<Utf8> str) {
+    final Pointer<Uint8> array = str.cast();
+    int count = 0x1000;
+    Uint8List string = array.asExternalTypedData(count: count);
+    int i = 0;
+    for (; string[i] != 0; ++i) {
+      if (i == count) {
+        count *= 2;
```

Comment:
If I read this correctly, then:
* you increment `i` so that it reaches `count`.
* you read `string[i]` and compare it to 0
* then you check whether `i` is `count`.

That should give an out-of-bounds error on the read before checking. 

You could move the `++i` into the body of the loop, so it's executed before the `i == count` check. Then the loop could just be a `while` loop.

Reply:
Done.